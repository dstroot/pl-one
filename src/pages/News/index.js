import React, { useEffect } from "react";
import "./styles.scss";

const News = () => {
  // Set the page title and position using the useEffect hook
  useEffect(() => {
    document.title = `Pacific Life â€¢ News`;
    window.scrollTo(0, 0);
  });

  return (
    <div>
      <h1>News!</h1>
    </div>
  );
};

export default News;
