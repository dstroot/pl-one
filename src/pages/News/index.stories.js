import React from "react";
import { storiesOf } from "@storybook/react";
import News from "../News";

storiesOf("News", module)
  .addParameters({
    // default for all stories in this book
    info: {
      inline: true,
      header: false,
      text: `This supports markdown!
      
      ~~~js
      console.log("hello");
      ~~~
      `
    }
  })
  .add("show News", () => <News />);
