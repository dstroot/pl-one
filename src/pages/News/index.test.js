import React from "react";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import News from "../News";

describe("News", () => {
  // mount the component
  let mountedComponent;
  const getMountedComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<News />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it("it should render", () => {
    let tree = create(<News />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("it should say News!", () => {
    const h1 = getMountedComponent()
      .find("h1")
      .first();

    expect(h1.text()).toContain("News!");
  });
});
