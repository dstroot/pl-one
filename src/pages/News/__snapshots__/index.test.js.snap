// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News it should render 1`] = `
<div>
  <h1>
    News!
  </h1>
</div>
`;
